force-app/main/default/classes/DMLOperations.cls

```diff
@@ -148,6 +174,16 @@ public with sharing class DMLOperations {
         // Loop through each Opportunity name
             // Create an Opportunity with the given name
             // Add the Opportunity to the list
+        
+        Account acc = new Account(Name = accountName);
+        upsert acc;
+        
+        List<Opportunity> newOpps = new List<Opportunity>();
+        for (String oppName : oppNames) {
+            Opportunity newOpp = new Opportunity(Name = oppName, AccountId = acc.Id, StageName = 'Prospecting', CloseDate = Date.today().addMonths(1));
+            newOpps.add(newOpp);
+        }
+        upsert newOpps;
```

Comment:
The tests pass for this because it is really is not testing the full realistic scenario. Can you try and check if the account is already existing, use that account, and check if the opportunities already exist before inserting them?

Reply:
Refactored this. Not sure if the tostring().contains on the List<Opportunties> was the best approach. Felt like there should be something better but couldn't find it.

Reply:
@leekendall This is interesting. The method `toString()` seems to work in this case, but it's not advisable. Instead, I would create a new list of `existingOppNames` (by looping over `oppsExisting`). And I would check the `oppNames` against `existingOppNames`. Does that make sense?

Reply:
toString in this case wont work on very long Lists because it usually it truncates automatically.